compiler/modules/front.syntax/token.bal

```diff
@@ -245,10 +245,25 @@ class Tokenizer {
         self.mode = m;
     }
 
-    function currentPos() returns Position {
+    function currentStartPos() returns Position {
         return createPosition(self.lineIndex, self.tokenStartCodePointIndex);
     }
 
+    function currentEndPos() returns Position {
+        Token? tok = self.current();
```

Comment:
This should be done using self.codePointIndex. Not sure if anything else is needed (don't think there is). If there is, current() should set it up and it should be returned here.

Reply:
fixed